yaetos/etl_utils.py

```diff
@@ -560,90 +827,157 @@ def save_metadata(self, elapsed):
+        )
 
     def push_to_kafka(self, output, types):
-        """ Needs to be overriden by each specific job."""
+        """Needs to be overriden by each specific job."""
         raise NotImplementedError
 
     def send_msg(self, msg, recipients=None):
-        """ Sending message to recipients (list of email addresse) or, if not specified, to yml 'owners'.
+        """Sending message to recipients (list of email addresse) or, if not specified, to yml 'owners'.
         Pulling email sender account info from connection_file."""
         if not recipients:
-            recipients = self.jargs.merged_args.get('owners')
+            recipients = self.jargs.merged_args.get("owners")
         if not recipients:
-            logger.error("Email can't be sent since no recipient set in {}, .\nMessage : \n{}".format(self.jargs.job_param_file, msg))
+            logger.error(
+                "Email can't be sent since no recipient set in {}, .\nMessage : \n{}".format(
+                    self.jargs.job_param_file, msg
+                )
+            )
             return None
 
-        creds = Cred_Ops_Dispatcher().retrieve_secrets(self.jargs.storage, creds=self.jargs.connection_file)
+        creds = Cred_Ops_Dispatcher().retrieve_secrets(
+            self.jargs.storage, creds=self.jargs.connection_file
+        )
         creds_section = self.jargs.email_cred_section
 
-        sender_email = creds.get(creds_section, 'sender_email')
-        password = creds.get(creds_section, 'password')
-        smtp_server = creds.get(creds_section, 'smtp_server')
-        port = creds.get(creds_section, 'port')
+        sender_email = creds.get(creds_section, "sender_email")
+        password = creds.get(creds_section, "password")
+        smtp_server = creds.get(creds_section, "smtp_server")
+        port = creds.get(creds_section, "port")
 
         for recipient in recipients:
             send_email(msg, recipient, sender_email, password, smtp_server, port)
-            logger.info('Email sent to {}'.format(recipient))
+            logger.info("Email sent to {}".format(recipient))
 
     def send_job_failure_email(self, error_msg):
-        message = """Subject: [Data Pipeline Failure] {name}\n\nA Data pipeline named '{name}' failed.\nError message:\n{error}\n\nPlease check logs in AWS.""".format(name=self.jargs.job_name, error=error_msg)
+        message = """Subject: [Data Pipeline Failure] {name}\n\nA Data pipeline named '{name}' failed.\nError message:\n{error}\n\nPlease check logs in AWS.""".format(
+            name=self.jargs.job_name, error=error_msg
+        )
         self.send_msg(message)
 
+    @staticmethod
+    def format_datetime(wiki_dt_str: str) -> str:
+        """
+        Converts a string date to a datetime to then parse to a target format.
+        """
+        wiki_dt = datetime.strptime(wiki_dt_str, "%Y%m%d%H%M%S")
+        return wiki_dt.strftime("%Y-%m-%d %H:%M:%S")
```

Comment:
Yes, much better than the code I had ! I guess I didn't know about this option at the time.